### PR TITLE
Update fmt (to 11.0.2) and spdlog (to 1.14.1).

### PR DESCRIPTION
**Pull request by jameslamb**
Contributes to:

* https://github.com/rapidsai/build-planning/issues/54
* https://github.com/rapidsai/build-planning/issues/56
* https://github.com/rapidsai/rapids-cmake/issues/387

Replaces https://github.com/rapidsai/rmm/pull/1544

Now that most of `conda-forge` has been updated to `fmt >=11.0.1,<12` and `spdlog>=1.14.1,<1.15` (https://github.com/rapidsai/build-planning/issues/56#issuecomment-2334281452), we're attempting to upgrade RAPIDS to similar versions of those libraries.

This impoves the likelihood that RAPIDS will be installable alongside newer versions of its dependencies and complementary packages on conda-forge.

## Notes for Reviewers

This PR is testing changes made in https://github.com/rapidsai/rapids-cmake/pull/689.
It shouldn't be merged until those `rapids-cmake` changes are merged and any testing-specific details have been removed.

### CODE REVIEW

File: dependencies.yaml [hunk 1]
```diff
@@ -111,8 +111,8 @@ dependencies:
         packages:
           - c-compiler
           - cxx-compiler
-          - fmt>=10.1.1,<11
-          - spdlog>=1.12.0,<1.13
+          - fmt>=11.0.2,<12
+          - spdlog>=1.14.1,<15
```
Comment (bdice) [inline dependencies.yaml:115]:
```suggestion
          - spdlog>=1.14.1,<1.15
```

File: conda/recipes/librmm/conda_build_config.yaml [hunk 2]
```diff
@@ -14,10 +14,10 @@ cmake_version:
   - ">=3.26.4,!=3.30.0"
 
 fmt_version:
-  - ">=10.1.1,<11"
+  - ">=11.0.2,<12"
 
 spdlog_version:
-  - ">=1.12.0,<1.13"
+  - ">=1.14.1,<15"
```
Comment (bdice) [inline conda/recipes/librmm/conda_build_config.yaml:20]:
```suggestion
  - ">=1.14.1,<1.15"
```

File: dependencies.yaml (hunk 1 above)
Comment (jameslamb) [reply dependencies.yaml:115]: 🙈 thank you

File: conda/environments/all_cuda-118_arch-x86_64.yaml [hunk 3]
```diff
@@ -35,7 +35,7 @@ dependencies:
 - python>=3.10,<3.13
 - rapids-build-backend>=0.3.0,<0.4.0.dev0
 - scikit-build-core >=0.10.0
-- spdlog>=1.12.0,<1.13
+- spdlog>=1.14.1,<15
```
Comment (bdice) [inline conda/environments/all_cuda-118_arch-x86_64.yaml:38]:
```suggestion
- spdlog>=1.14.1,<1.15
```

File: conda/environments/all_cuda-125_arch-x86_64.yaml [hunk 4]
```diff
@@ -34,7 +34,7 @@ dependencies:
 - python>=3.10,<3.13
 - rapids-build-backend>=0.3.0,<0.4.0.dev0
 - scikit-build-core >=0.10.0
-- spdlog>=1.12.0,<1.13
+- spdlog>=1.14.1,<15
```
Comment (bdice) [inline conda/environments/all_cuda-125_arch-x86_64.yaml:37]:
```suggestion
- spdlog>=1.14.1,<1.15
```

File: cmake/thirdparty/get_spdlog.cmake [hunk 5]
```diff
@@ -16,19 +16,10 @@
...
-      GLOBAL_TARGETS spdlog spdlog_header_only
-      NAMESPACE spdlog::)
-    include("${rapids-cmake-dir}/export/find_package_root.cmake")
-    rapids_export_find_package_root(BUILD spdlog [=[${CMAKE_CURRENT_LIST_DIR}]=]
-                                    EXPORT_SET rmm-exports)
-  endif()
+  rapids_cpm_spdlog(
+    FMT_OPTION "EXTERNAL_FMT_HO"
+    INSTALL_EXPORT_SET rmm-exports
+    BUILD_EXPORT_SET rmm-exports)
```
Comment (jameslamb) [inline cmake/thirdparty/get_spdlog.cmake:22]:
```suggestion
    BUILD_EXPORT_SET rmm-exports)
  rapids_export_package(BUILD spdlog rmm-exports)
```

File: cmake/thirdparty/get_spdlog.cmake [hunk 6]
```diff
@@ -16,19 +16,11 @@
...
-  if(spdlog_ADDED)
-    rapids_export(
-      BUILD spdlog
-      EXPORT_SET spdlog
-      GLOBAL_TARGETS spdlog spdlog_header_only
-      NAMESPACE spdlog::)
-    include("${rapids-cmake-dir}/export/find_package_root.cmake")
-    rapids_export_find_package_root(BUILD spdlog [=[${CMAKE_CURRENT_LIST_DIR}]=]
-                                    EXPORT_SET rmm-exports)
-  endif()
```
Comment (jameslamb) [inline cmake/thirdparty/get_spdlog.cmake:31]: Context for these `get_spdlog.cmake` changes: https://github.com/rapidsai/rapids-cmake/issues/387#issuecomment-2077550701

File: rapids_config.cmake [hunk 7]
```diff
@@ -25,6 +25,8 @@ else()
       "Could not determine RAPIDS version. Contents of VERSION file:\n${_rapids_version_formatted}")
 endif()
 
+set(rapids-cmake-repo jameslamb/rapids-cmake)
+set(rapids-cmake-branch fmt-and-spdlog)
```
Comment (jameslamb) [inline rapids_config.cmake:29]:
```suggestion
```

TODO: remove this right before merging

Comment (jameslamb) [reply rapids_config.cmake]: removed in https://github.com/rapidsai/rmm/pull/1678/commits/4135b68d5cc1673636ed8a114b2275734146c229

**Pull request merged**